src/dmd/dsymbolsem.d

```diff
@@ -5501,6 +5542,8 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         if (0 &&          // deprecation disabled for now to accommodate existing extensive use
             cldec.storage_class & STC.scope_)
             deprecation(cldec.loc, "`scope` as a type constraint is deprecated.  Use `scope` at the usage site.");
+
+        checkInterfaceImplementations(cldec);
```

Comment:
I'm not sure if this is the best place to check this. I've tried to come up with code snippets that would break this, but have failed so far.